### PR TITLE
Change route

### PR DESCRIPTION
**Pull request by EugeneTeu**

### CODE REVIEW

File: frontend/src/pages/dashboard/user-offer/offer-item.tsx [hunk 1]
```diff
@@ -61,7 +58,7 @@ const OfferItem: FC<Props> = ({ offer, fetch }) => {
 	<Spin spinning={loading}>
     <Card
       className={s.cardStyle}
-      actions={[<GoogleOutlined onClick={handleCancel} />]}
+      actions={[<Button onClick={handleCancel} >Cancel <CloseCircleFilled/> </Button>]}
```
Comment (pohlinwei) [inline frontend/src/pages/dashboard/user-offer/offer-item.tsx:64]: HAHA. You found it (:

**Review by pohlinwei — APPROVED** [review]

File: frontend/src/logic/requestlogic.ts [hunk 2]
```diff
@@ -107,21 +107,28 @@ export const getRequests: (
...
+          const request = await requestConverter.fromFirestore(requestSnapshot);
+          results.push(request);
+        } catch (e) {
+          return console.error(
+            `Encountered error while retrieving request: ${e.message}`
+          );
+        }
+      })
+    );
+    return results.filter((x) => x.task.status === Status.DONE);
```
Comment (pohlinwei) [inline frontend/src/logic/requestlogic.ts]: Are you sure that you only need to filter for `DONE`? 😅

File: frontend/src/pages/dashboard/market-place/components/main-offer/index.tsx [hunk 3]
```diff
@@ -144,8 +144,8 @@ const MainOffer: FC<Props> = ({ uid , email}) => {
 						{modalOffer?.shopLocation}
 					</Paragraph>
 					<Paragraph>
-						<Text strong={true}>Doing Offer: </Text>
-						{modalOffer?.doerName}
+						<Text strong={true}>Status: </Text>
```
Comment (pohlinwei) [inline frontend/src/pages/dashboard/market-place/components/main-offer/index.tsx:147]: Why did you choose to display status instead? 😅 I thought that all offers' status will be open by default, won't they?

File: frontend/src/pages/dashboard/market-place/components/main-request/index.tsx [hunk 4]
```diff
@@ -165,7 +165,7 @@ const MainRequest: FC<Props> = ({uid, email}) => {
 				okText="fufil request?"
 			>
 				<Typography>
-					<Title>{modalReq?.task.payerName}</Title>
+					<Title>{modalReq?.task.payerUid}</Title>
```
Comment (pohlinwei) [inline frontend/src/pages/dashboard/market-place/components/main-request/index.tsx:168]: Is this a placeholder? We will be replacing this with its display name, won't we?

File: frontend/src/pages/dashboard/user-history/index.tsx [hunk 5]
```diff
@@ -26,7 +26,7 @@ const UserHistory: FC<DashboardCompProps> = ({ userUid }) => {
         setLoading(true);
         const resOffer = await getOffers(userUid);
         const resTask = await getTasks(userUid);
-        const resRequest = await getRequests(userUid);
+        const resRequest = await getRequests(userUid, Status.DONE);
```
Comment (pohlinwei) [inline frontend/src/pages/dashboard/user-history/index.tsx:29]: So history only shows the completed requests? 😅 What about those that were cancelled?

File: frontend/src/pages/dashboard/user-history/index.tsx [hunk 6]
```diff
@@ -75,7 +75,7 @@ const UserHistory: FC<DashboardCompProps> = ({ userUid }) => {
                 dataSource={requests}
                 renderItem={(item) => (
                   <List.Item>
-                    <Card title={item.task.payerName} style={{ width: "100%" }}>
+                    <Card title={ `Payer ID: ${item.task.payerUid}`} style={{ width: "100%" }}>
```
Comment (pohlinwei) [inline frontend/src/pages/dashboard/user-history/index.tsx:78]: Is this another placeholder?

File: frontend/src/pages/dashboard/market-place/components/main-request/index.tsx (hunk 4 above)
Comment (EugeneTeu) [reply frontend/src/pages/dashboard/market-place/components/main-request/index.tsx:168]: Oh yea yes we will

File: frontend/src/pages/dashboard/user-history/index.tsx (hunk 5 above)
Comment (EugeneTeu) [reply frontend/src/pages/dashboard/user-history/index.tsx:29]: Yea could I request for a history api? Right now was using like a placeholder method with GetRequests and all that

File: frontend/src/pages/dashboard/user-history/index.tsx (hunk 6 above)
Comment (EugeneTeu) [reply frontend/src/pages/dashboard/user-history/index.tsx:78]: Yes as mentioned will change to user display name

**Comment by EugeneTeu** [comment]
@pohlinwei thanks for checking, currently can merge and u can work on ur side , my changes after this should be comestic

File: frontend/src/pages/dashboard/user-history/index.tsx (hunk 5 above)
Comment (pohlinwei) [reply frontend/src/pages/dashboard/user-history/index.tsx:29]: Okay!

**Comment by pohlinwei** [comment]
No worries! (:

File: frontend/src/pages/dashboard/user-history/index.tsx (hunk 5 above)
Comment (pohlinwei) [reply frontend/src/pages/dashboard/user-history/index.tsx:29]: Sorry, just to clarify, 'history' should show the user's entire history, shouldn't it? 😅 If so, do we need the status? Or shall I order them by status? In particular, I will return `{ tasks: {done: [...], cancelled: [..], pending: [...], ...}, offers: {open: [...], ...}, requests: {done: [...], ...}}`.

Comment (EugeneTeu) [reply frontend/src/pages/dashboard/user-history/index.tsx:29]: Can be 3 endpoints or one actually dowsnt matter haha, Yeap the structure sounds good

**Review by leafgecko — APPROVED** [review]

**Pull request merged**